src/main/java/internetshop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package internetshop.dao.impl;
+
+import internetshop.dao.ShoppingCartDao;
+import internetshop.db.Storage;
+import internetshop.lib.Dao;
+import internetshop.model.ShoppingCart;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingcart) {
+        Storage.addShoppingCart(shoppingcart);
+        return shoppingcart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long shoppoingcartId) {
+        return Optional.of(Storage.shoppingCarts
+                .stream()
+                .filter(shoppingcart -> shoppingcart.getId().equals(shoppoingcartId))
+                .findFirst()
+                .orElseThrow(NoSuchElementException::new));
```

Comment:
```suggestion
                .findFirst();
```
ми повертаємо Опшинал, тому тут не потрібно викидувати ексепшин